kola/tests/podman/podman.go

```diff
@@ -0,0 +1,352 @@
+func init() {
+	register.Register(&register.Test{
+		Run:         podmanBaseTest,
+		ClusterSize: 1,
+		Name:        `podman.base`,
+		Distros:     []string{"rhcos"},
+	})
+	register.Register(&register.Test{
+		Run:              podmanWorkflow,
+		ClusterSize:      1,
+		Name:             `podman.workflow`,
+		ExcludePlatforms: []string{"qemu"}, // Requires internet for pulling nginx
+		Distros:          []string{"rhcos"},
+	})
+	register.Register(&register.Test{
+		Run:         podmanNetworkTest,
+		ClusterSize: 2,
+		Name:        `podman.network`,
+		Distros:     []string{"rhcos"},
+	})
+}
+
+// simplifiedContainerPsInfo represents a container entry in podman ps -a
+type simplifiedContainerPsInfo struct {
+	Id     string `json:"id"`
+	Image  string `json:"image"`
+	Status string `json:"status"`
+}
+
+// simplifiedPsInfo represents the results of podman ps -a
+type simplifiedPsInfo struct {
+	containers []simplifiedContainerPsInfo
+}
+
+// simplifiedPodmanInfo represents the results of podman info
+type simplifiedPodmanInfo struct {
+	Store struct {
+		GraphDriverName string `json:"GraphDriverName"`
+		GraphRoot       string `json:"GraphRoot"`
+	}
+}
+
+func getSimplifiedPsInfo(c cluster.TestCluster, m platform.Machine) (simplifiedPsInfo, error) {
+	target := simplifiedPsInfo{}
+	psJson, err := c.SSH(m, `sudo podman ps -a --format json`)
+
+	if err != nil {
+		return target, fmt.Errorf("could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(psJson, &target.containers)
+
+	if err != nil {
+		return target, fmt.Errorf("could not unmarshal info %q into known json: %v", string(psJson), err)
+	}
+	return target, nil
+}
+
+// Returns the result of podman info as a simplifiedPodmanInfo
+func getPodmanInfo(c cluster.TestCluster, m platform.Machine) (simplifiedPodmanInfo, error) {
+	target := simplifiedPodmanInfo{}
+
+	pInfoJson, err := c.SSH(m, `sudo podman info --format json`)
+	if err != nil {
+		return target, fmt.Errorf("Could not get info: %v", err)
+	}
+
+	err = json.Unmarshal(pInfoJson, &target)
+	if err != nil {
+		return target, fmt.Errorf("Could not unmarshal info %q into known JSON: %v", string(pInfoJson), err)
+	}
+	return target, nil
+}
+
+// make a podman container out of binaries on the host
+func genPodmanContainer(c cluster.TestCluster, m platform.Machine, name string, binnames []string) {
+	cmd := `tmpdir=$(mktemp -d); cd $tmpdir; echo -e "FROM scratch\nCOPY . /" > Dockerfile;
+	        b=$(which %s); libs=$(sudo ldd $b | grep -o /lib'[^ ]*' | sort -u);
+	        sudo rsync -av --relative --copy-links $b $libs ./;
+	        sudo podman build -t %s .`
+
+	c.MustSSH(m, fmt.Sprintf(cmd, strings.Join(binnames, " "), name))
+}
+
+func podmanBaseTest(c cluster.TestCluster) {
+	c.Run("info", podmanInfo)
+	c.Run("resources", podmanResources)
+	c.Run("network", podmanNetworksReliably)
+}
+
+// Test: Run basic podman commands
+func podmanWorkflow(c cluster.TestCluster) {
```

Comment:
Ideally we'd like to keep tests self-contained, having unreported sub tests that depend on each other isn't optimal. In this specific case I would be okay with this for now.

It does start me thinking about adding some sort of `failfast` setting inside a `TestCluster` though. If we had a feature like that we could pretty easily wrap each comment marked block in a `c.Run` and let the harness keep track of whether a previous sub test has failed (immediately marking any sub tests following a failed sub test as failed without running them, keeping the functionality of this code).

Reply:
I have up a PR for a `FailFast` flag in #926 

That patch would allow you to wrap each commented test block in a `c.Run` and then if one sub-test fails all future sub-tests would be skipped.